File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/entity/AssetHistory.java
package com.NBE3_4_2_Team4.domain.asset.main.entity;

import com.NBE3_4_2_Team4.domain.member.member.entity.Member;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class AssetHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long amount;

    @Column(nullable = false)
    private String correlationId;  // 연관 그룹 ID

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne
    private Member member;

    @ManyToOne
    @JoinColumn
    private Member counterMember;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AssetCategory assetCategory;

    //Cash or Point
    @Column(nullable = false)
    private AssetType assetType;
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/entity/AssetHistory.kt
package com.NBE3_4_2_Team4.domain.asset.main.entity

import com.NBE3_4_2_Team4.domain.member.member.entity.Member
import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@EntityListeners(
    AuditingEntityListener::class
)
class AssetHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @CreatedDate
    @Column(updatable = false)
    lateinit var createdAt: LocalDateTime

    @ManyToOne
    val member: Member

    @Column(nullable = false)
    val amount: Long

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val assetType: AssetType

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val assetCategory: AssetCategory

    @Column(nullable = false)
    val correlationId: String // 연관 그룹 ID

    @ManyToOne
    val adminAssetCategory: AdminAssetCategory?

    @ManyToOne
    @JoinColumn
    val counterMember: Member?

    constructor(
        member: Member,
        amount: Long,
        assetType: AssetType,
        assetCategory: AssetCategory,
        correlationId: String,
        adminAssetCategory: AdminAssetCategory? = null,
        counterMember: Member? = null
    ) {
        this.member = member
        this.amount = amount
        this.assetType = assetType
        this.assetCategory = assetCategory
        this.correlationId = correlationId
        this.adminAssetCategory = adminAssetCategory
        this.counterMember = counterMember
    }
}
